backend/support/cog_over_w_audit.py

```diff
@@ -0,0 +1,289 @@
+
+    try:
+        audit_model = apps.get_model("audit.Audit")
+        cognizant_agency = (
+            audit_model.objects.filter(
+                Q(audit_year__in=query_years) & (query_subsection)
+            )
+            .exclude(cognizant_agency__isnull=True)
+            .exclude(cognizant_agency__exact="")
+            .order_by("-audit_year")
+            .values_list("cognizant_agency", flat=True)[:]
+        )
+        if len(cognizant_agency) >= 1:
+            return cognizant_agency[0]
+    except audit_model.DoesNotExist:
+        cognizant_agency = None
+    return cognizant_agency
+
+
+def get_base_audit(ein, uei, base_year):
+    audit_model = apps.get_model("audit.Audit")
+    audits = audit_model.objects.filter(
+        Q(auditee_ein=ein), Q(auditee_uei=uei), Q(audit_year=base_year)
+    )
+
+    if len(audits) != 1:
+        return (len(audits), 0, None)
+    audit = audits[0]
+    amt = audit.audit["federal_awards"]["total_amount_expended"]
+    return (1, amt, audit.report_id)
+
+
+def get_base_cfdas(report_id):
+    audit_model = apps.get_model("audit.Audit")
+    audit = audit_model.objects.get(report_id=report_id)
+    cfdas = audit.audit["federal_awards"]["awards"]
+
+    if len(cfdas) == 0:
+        return None
+    baseline_cfdas = []
+    for row in cfdas:
+        baseline_cfdas.append(
+            (
+                row["program"]["federal_agency_prefix"],
+                row["program"]["amount_expended"],
+                row["direct_or_indirect_award"]["is_direct"],
+            )
+        )
+    return baseline_cfdas
+
+
+def calc_cfda_amounts(cfdas):
+    total_amount_agency = defaultdict(lambda: 0)
+    total_da_amount_agency = defaultdict(lambda: 0)
+    for cfda in cfdas:
+        agency = cfda[0]
+        amount = cfda[1] or 0
+        direct = cfda[2]
+        total_amount_agency[agency] += amount
+        if direct == "Y":
+            total_da_amount_agency[agency] += amount
+    return (
+        prune_dict_to_max_values(total_amount_agency),
+        prune_dict_to_max_values(total_da_amount_agency),
+    )
+
+
+def prune_dict_to_max_values(data: dict):
+    """
+    prune_dict_to_max_values({"a": 5, "b": 10, "c": 10}) => {"b": 10, "c": 10}
+    """
+    if len(data) == 0:
+        return data
+
+    pruned_dict = {}
+    max_value = max(data.values())
+    for key, value in data.items():
+        if value == max_value:
+            pruned_dict[key] = value
+    return pruned_dict
+
+
+def record_cog_assignment(report_id, user, cognizant_agency):
+    """
+    To be unvoked by app to persist the computed cog agency
+    """
+    CognizantAssignment(
+        report_id=report_id,
+        cognizant_agency=cognizant_agency,
+        assignor_email=user.email,
+    ).save()
+
+
+def _get_cog_over(audit):
+    """
+    Function that the FAC app uses when a submission is completed and cog_over needs to be assigned.
+    """
+    if "federal_awards" not in audit.audit.keys():
```

Comment:
```suggestion
    if not audit.audit.get("federal_awards"):
```

This should help if the key is there, but it's empty.